exports/dataset.cpp

```diff
@@ -624,9 +626,47 @@ PYBIND11_MODULE(dataset, m) {
           "numpy", &as_py_array_t_variant<Variable, double, float, int64_t,
                                           int32_t, char, bool>)
       .def_property_readonly("data", &as_VariableView::get<Variable>)
+      .def(py::self + py::self, py::call_guard<py::gil_scoped_release>())
+      .def(py::self + float(), py::call_guard<py::gil_scoped_release>())
+      .def("__add__",
```

Comment:
Have you verified that this variant is actually used? `pybind11` tries overloads in order of appearance, so it is likely that the one two lines above will match, doing an unnecessary conversion.

Reply:
Runtime checked. This float and VariableSlice overloads are called as expected.

Reply:
Ah, wait. I see that this is probably due to an inconsistency. We have

```cpp
  py::implicitly_convertible<DatasetSlice, Dataset>();
```
but the same is missing for `Variable`. If you add that, can you get rid of some of the overloads I had mentioned in an earlier comment?